### PR TITLE
Deny submissions for eliminated teams

### PR DESCRIPTION
**Pull request by Coderbora**

### CODE REVIEW

File: api/helpers/results.ts [hunk 1]
```diff
@@ -26,6 +28,14 @@ export interface JudgeCorrel {
     correl: number;
 }
 
+export interface RoundResults {
```
Comment (milaneyia) [inline api/helpers/results.ts]: this already exists https://github.com/milaneyia/obwc/blob/main/shared/integration.ts#L42 try to use that one

File: api/models/Team.ts [hunk 2]
```diff
@@ -42,4 +48,14 @@ export class Team extends BaseEntity {
     @OneToMany(() => Submission, (submissions) => submissions.team)
     submissions!: Submission[];
 
+    async getElimination (round: Round): Promise<EliminationDetails | undefined> {
+        const mapperResults = await getRoundResults(round.id, JUDGING_TYPE.Mappers, ResultsScope.User);
+        const playersResults = await getRoundResults(round.id, JUDGING_TYPE.Players, ResultsScope.User);
+
+        const mappingEliminated = mapperResults.teamsScores.filter(teams => teams.isEliminated).some(teams => teams.team.id === this.id);
```
Comment (milaneyia) [inline api/models/Team.ts]: `mapperResults.teamsScores.some(ts => ts.team.id === this.id && ts.isEliminated` is the same and just 1 loop?

File: api/models/Team.ts [hunk 3]
```diff
@@ -42,4 +48,14 @@ export class Team extends BaseEntity {
     @OneToMany(() => Submission, (submissions) => submissions.team)
     submissions!: Submission[];
 
+    async getElimination (round: Round): Promise<EliminationDetails | undefined> {
+        const mapperResults = await getRoundResults(round.id, JUDGING_TYPE.Mappers, ResultsScope.User);
```
Comment (milaneyia) [inline api/models/Team.ts]: `isEliminated` is based from round 1 results (as later rounds wont have more than 1 team per country), so u could just do `getRoundResults(1, JUDGING_TYPE.Mappers, ResultsScope.User);` or query all rounds, sort by results date and take the first one up to u

File: api/models/Team.ts [hunk 4]
```diff
@@ -42,4 +48,14 @@ export class Team extends BaseEntity {
     @OneToMany(() => Submission, (submissions) => submissions.team)
     submissions!: Submission[];
 
+    async getElimination (round: Round): Promise<EliminationDetails | undefined> {
```
Comment (milaneyia) [inline api/models/Team.ts]: shouldnt this return `Promise<EliminationDetails>`? it wont ever be undefined i believe?

File: api/routes/submissions.ts [hunk 5]
```diff
@@ -58,6 +58,13 @@ submissionsRouter.post('/', getCurrentRound(RoundScope.Submission), koaBody({
     const oszFile = ctx.request.files?.oszFile as FFile | undefined;
     const information = validator.trim(ctx.request.body.information);
 
+    const checkRound = await Round.currentRound(RoundScope.Results).getOne();
```
Comment (milaneyia) [inline api/routes/submissions.ts]: see about what i wrote in getElimination()

File: shared/models.ts [hunk 6]
```diff
@@ -98,6 +103,8 @@ export interface Team {
     users: User[];
     invitations: User[];
     wasConfirmed: boolean;
+
+    getElimination(): Promise<EliminationDetails | undefined>;
```
Comment (milaneyia) [inline shared/models.ts]: these interfaces are just for client side responses, so this is unneeded

**Review by milaneyia — CHANGES_REQUESTED** [review]
using rounds other than the first one for checking elimination state will be wrong
rest looks ok to me

File: api/models/Team.ts (hunk 3 above)
Comment (Coderbora) [reply api/models/Team.ts]: I wasn't sure about country eliminations but as you said, eliminations are based on the first round so I'll do it for first round.

**Pull request merged**